lib/libutils/isoc/bget_malloc.c

```diff
@@ -609,7 +613,11 @@ void *mdbg_malloc(const char *fname, int lineno, size_t size)
 {
 	struct mdbg_hdr *hdr;
 
+#if defined(ARM64)
+	COMPILE_TIME_ASSERT(sizeof(struct mdbg_hdr) == sizeof(uint32_t) * 6);
```

Comment:
If the `__alignof(uintptr_t)` is the *fundamental alignment requirement* you mentioned. It seems like the required alignment is also 8 for arm64.